src/org/connection.ts

```diff
@@ -342,19 +342,21 @@ export class Connection<S extends Schema = Schema> extends JSForceConnection<S>
    */
   public async autoFetchQuery<T extends Schema = S>(
     soql: string,
-    queryOptions: Partial<QueryOptions> = {}
+    queryOptions: Partial<QueryOptions & { tooling: boolean }> = { tooling: false }
   ): Promise<QueryResult<T>> {
     const config: ConfigAggregator = await ConfigAggregator.create();
     // take the limit from the calling function, then the config, then default 10,000
     const maxFetch: number =
       (config.getInfo(OrgConfigProperties.ORG_MAX_QUERY_LIMIT).value as number) || queryOptions.maxFetch || 10000;
 
+    const tooling = queryOptions?.tooling;
+    delete queryOptions.tooling;
+
```

Comment:
```suggestion
    const { tooling, ...queryOptionsWithoutTooling }  = queryOptions;
```

and then use the new constant below.

logic: generally avoid mutating parameters passed to functions (especially objects passed by reference) because whoever calls this may not be done with the object and shouldn't be aware that you're changing it.